### PR TITLE
family-dns: add new package

### PR DESCRIPTION
**Pull request by farmergreg**
Maintainer: me
Compile tested: amd64, and ath79
Run tested: amd64 server, tp-link archer a7 HWv5

Description:
This package configures OpenWrt routers to quickly and easily block adult websites.
The default settings are designed to be appropriate for most businesses,
schools and families.

Signed-off-by: Gregory L. Dietsche <gregory.dietsche@cuw.edu>

### CODE REVIEW

File: net/family-dns/files/family-dns-update [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#!/bin/sh
```
Comment (neheb) [inline net/family-dns/files/family-dns-update:1]: Please run through shellcheck. I see bashisms.

File: net/family-dns/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2020 Gregory L. Dietsche <Gregory.Dietsche@cuw.edu>
+# This is free software, licensed under the MIT License
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=family-dns
+PKG_VERSION:=1.0.0
+PKG_RELEASE:=0
```
Comment (neheb) [inline net/family-dns/Makefile:10]: Set to 1.

**Comment by farmergreg** [comment]
Thanks for the review @neheb!
I've updated PKG_RELEASE=1 and also ran family-dns-update through shellcheck and fixed those errors. I also noticed an issue with the error checking for  `echo "$dns" is not supported.` and added this to fix the problem:
```
uci revert network
redirect_dns=0
```

**Comment by dibdot** [comment]
I'm not sure if we really need a package that sets only DNS server entries basically. But if you're going to support this, it's OK for me.

Some remarks on the code after a first sight:
- please use '$(...)' notation instead of legacy backticks
-  please use double quotes to prevent globbing and word splitting
- the ROT13 encoding/encryption of porn domains is at least superfluous
- the 'count' variable appears to be unused, please remove it

**Comment by dibdot** [comment]
I'm not sure if we really need a package that sets only DNS server entries basically. But if you're going to support this, it's OK for me.

Some remarks on the code after a first sight:
- please use '$(...)' notation instead of legacy backticks
-  please use double quotes to prevent globbing and word splitting
- the ROT13 encoding/encryption of porn domains is at least superfluous
- the 'count' variable appears to be unused, please remove it

**Comment by farmergreg** [comment]
Thanks for the feedback, @dibdot . I've made the changes you requested. I'm happy to maintain the package indefinitely! It's part of a larger project I'm working on. Having the package upstream so it can be quickly/easily installed by an end-user with little technical knowledge, and also being able to generate custom images easily with image-builder are my goals.

**Comment by dibdot** [comment]
@farmergreg Thanks!
The ROT13 oddity is still in there. It just makes no sense to work with it, security by obscurity does not work.

**Comment by farmergreg** [comment]
@dibdot , ROT13 wasn't meant as a security measure. I don't like seeing the domain names when I run the tests. I'm open to removing it, but would prefer that it stay for now... I'm probably the only one who will ever run those tests anyway :) 
I'm in the process of contacting those services about canary domains and/or better ways to detect that they're configured correctly. At some point I hope to have a patch for the test script to that effect...

Thanks Again!
Greg

**Comment by dibdot** [comment]
> I'm in the process of contacting those services about canary domains

Then my best guess would be 'www.playboy.com', not really porn but in every filter/dns provider for sure.

**Comment by farmergreg** [comment]
Unfortunately, it doesn't have an AAAA record, so it won't work for the IPv6 portion of the tests.

**Comment by farmergreg** [comment]
@dibdot, @neheb I've removed the rot13 code. I think this resolves all the code review issues. The most recently pushed code is run tested on both an amd64 openwrt router, and on a tp-link archer a7 HWv5 (ath79). Thanks!

**Pull request merged**